sync/syncer.go

```diff
@@ -351,6 +361,15 @@ func (s *Syncer) ATXs(mb *types.Block) (atxs []*types.ActivationTx, associated *
 		for out := range output {
 			ntxs := out.([]types.ActivationTx)
 			for _, atx := range ntxs {
+				err := s.SyncPoetProof(atx.GetPoetProofRef())
+				if err != nil {
+					if types.IsProcessingError(err) {
+						s.Error("error while validating PoET proof: %v", err)
+					} else {
+						s.Warning("received ATX with invalid or missing PoET proof: %v", err)
+					}
+					return nil, nil, fmt.Errorf("a referenced PoET proof is missing or invalid: %v", err)
```

Comment:
IMO we shouldn't halt on the first failure. Instead, we should fetch all ATXs, store those who failed and print all of them. It will ease the debugging

Reply:
I'm not sure it will ease debugging. I think it could cause weird situations that are harder to debug, e.g.:

We try to fetch the PoET proof of an ATX and fail. We keep that ATX ID for later and move on. We try to fetch the PoET proof for the next ATX, which happens to be the same one, essentially retrying. One of our neighbors now got this PoET proof and sends it to us. We then later print the list of ATXs that failed and it now doesn't make sense, since we see that we previously stored the PoET proof (after the retry).

Generally, I find that the simpler the logic, the easier the debugging.